### PR TITLE
feat: add filesystem performance stats fetching via stats API

### PR DESCRIPTION
**Pull request by sergeyberezansky**
### TL;DR
Adds a way to read per-filesystem I/O performance counters (throughput, IOPS, latency) from the Weka API. Nothing consumes it yet.

### What changed?
- Added `GetFilesystemPerformanceStats` to the Weka API client: fetches the last 60 seconds of read/write bytes, IOPS, and latency for a filesystem from the stats API.
- Added supporting types to parse the response and select the most recent sample.
- This is a building block only — no exported metric or driver behavior uses it in this PR.

### How to test?
No user-visible behavior yet; covered entirely by automated tests:
1. Run `go test ./pkg/wekafs/apiclient/...`.

### Why make this change?
Lays the groundwork for exposing per-volume performance metrics alongside the existing capacity metrics. It's landing on its own, ahead of the code that will wire it into the metrics server, so it can be reviewed and tested independently.

---

> [!NOTE]
> **Low Risk**
> New read-only API client code and types with no runtime integration; behavior is covered by unit tests and does not touch auth or data paths.
> 
> **Overview**
> Adds **per-filesystem I/O performance** fetching from the Weka **`stats`** API as a standalone building block (no metrics server or CSI behavior uses it in this PR).
> 
> The API client gains **`GetFilesystemPerformanceStats`**, which requests the last 60s of read/write bytes, IOPS, and latency for a filesystem, plus types to decode the reply and return the **newest sample** with a matching timestamp. **`FileSystem.GetFsIdAsInt`** parses `FSId<N>` ids for stats keys. Request shaping avoids two silent footguns: **`param[fS]=0`** is always sent (no `omitempty` on filesystem id), and only the six integer counters **`GetStats`** reads are requested so JSON decode does not fail on other stat shapes.
> 
> **`pkg/wekafs`** adds **`UsageStats`** and **`PvStats`** to bundle capacity and performance for future PV reporting. Coverage is in **`stats_test.go`** (sample ordering, per-FS filtering, query encoding).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6e3527473435211dd8858c93ed091efd9989e5e5. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by sergeyberezansky** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/723?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#723** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/723?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#724](https://github.com/weka/csi-wekafs/pull/724) <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/724?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#725](https://github.com/weka/csi-wekafs/pull/725) <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/725?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈 <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/723?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#722** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/722?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#826** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/826?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#803** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/803?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#802** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/802?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#801** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/801?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#800** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/800?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#799** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/799?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#798** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/798?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#797** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/797?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#796** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/796?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#795** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/795?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#791** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/791?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#786** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/786?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#701** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/701?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#787** <a href="https://app.graphite.com/github/pr/weka/csi-wekafs/787?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by sergeyberezansky** [comment]
> Moved from the PR description, which is now release-notes shaped.

This PR adds filesystem performance statistics collection to the CSI driver.

A new `GetFilesystemPerformanceStats` method on `ApiClient` fetches the last 60 seconds of I/O counters (read/write bytes, IOPS, and latency) for a given filesystem by querying the `stats` endpoint. Because that endpoint returns counters keyed by strings like `"READS[fS: 1]"` and groups them into time-bucketed samples, two pieces of supporting logic are introduced:

- `parseStatKey` splits those composite keys into a counter name and a numeric filesystem ID.
- `StatsResponse.GetStats` selects the most recent sample from the response and extracts only the counters belonging to the requested filesystem. The timestamp reported with the result is always taken from the same sample as the values, so the two cannot disagree.

To map a `FileSystem` object's string ID (e.g. `"FSId<3>"`) to the integer the stats endpoint uses, `GetFsIdAsInt` is added to `FileSystem`.

A new `PvStats` type bundles capacity (`UsageStats`) and performance (`PerfStats`) information about a volume, with either half allowed to be nil when only one was collected.

Tests cover sample ordering (oldest-first and newest-first), per-filesystem filtering, empty responses, key parsing edge cases, and ID conversion.

File: pkg/wekafs/apiclient/stats.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+type StatsRequest struct {
+	IntervalSeconds   int      `url:"interval,omitempty"`
+	Category          string   `url:"category,omitempty"`
+	Stats             []string `url:"stat,omitempty"`
+	ResolutionSeconds int      `url:"resolution_secs,omitempty"`
+	Accumulated       bool     `url:"accumulated,omitempty"`
+	ShowInternal      bool     `url:"show_internal,omitempty"`
+
+	Param struct {
+		FilesystemId int `url:"fS,omitempty"`
```
[the remark this answers is not in the document]
Comment (sergeyberezansky) [reply pkg/wekafs/apiclient/stats.go]: Confirmed and fixed — `omitempty` is gone from that field.

The chain is exactly as described: the API reports the first filesystem as `FSId<0>`, `GetFsIdAsInt` parses that to `0` and reserves `-1` for unknown, and the caller rejects only negatives. So `0` is a valid id that encodes to nothing, and the request goes out with no `param[fS]` at all — cluster-wide counters returned as though they belonged to that one filesystem.

Worth stating plainly because it is the worst shape of bug: not an error, just quietly wrong numbers for exactly one filesystem, and the one most likely to exist on any cluster.

The field is only ever set from an id that already passed the `>= 0` check, so there is nothing legitimate to omit. The test encodes the request with id 0 and asserts `param[fS]` is present; with `omitempty` restored it fails on `got "" (query "category=fs_stats&interval=60")`.
